### DISCUSSION TITLE
Cmd+s reverts back to old version

### DISCUSSION THREAD

**Discussion by psharp15** [zed-industries/zed#6535] [Feature Requests]
I'm using Zed 0.81.1 
When I'm trying to save a file, I do: Cmd+s, and instead of saving it, it reverts it back to the original version.

**Comment by psharp15** [comment]
But this is only for some of my files. What am I doing wrong?

**Reply by psharp15** [reply]
It stopped doing it for this one file. I'm not sure why that happened.

**Reply by psharp15** [reply]
It started doing it again on a different file. The first file I've attached is my code as I want it. And when I do "Cmd + s", it reformats it to this second image shown. 

![Screenshot 2023-04-13 at 3 35 10 PM](https://user-images.githubusercontent.com/66267349/231865050-1325c1be-3cc8-4ec4-a276-a3d3d5905506.png)

![Screenshot 2023-04-13 at 3 35 15 PM](https://user-images.githubusercontent.com/66267349/231865033-6cbaf72c-9abd-4d41-9236-7a917a037125.png)

**Comment by JosephTLyons** [comment]
Hey @psharp15, is the issue that its automatically formatting your code or is your file actually losing code changes that you made when you save it?

**Comment by psharp15** [comment]
Hi Mr. Lyons, 

Thank you for getting back to me! Unfortunately, it’s the first one: automatically formatting my code. I never manually reformatted my code in the way seen on the second image. It’s a gnuplot plotting script, but I’ve set the language format to be C++ to get the colorful syntax to apply. 

Just a quick test shows that this only happens when I set the language to C++. When I set it to C, CSS, or Python, it doesn’t reformat the code when I “Cmd + s”. So I’ll just avoid setting the language to C++, especially the not working on a C++ script. 

Thanks!

**Comment by ghost** [comment]
guys, why you still haven't fixed this bug? i am trying to save .c or .cpp files and zed changes all file formatting in awful way (from 4 spaces to 2 and etc)

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
